src/sanitize/sanitizeNumbers.spec.ts

```diff
@@ -279,6 +334,43 @@ describe('sanitizeNumbers', () => {
 		});
 
 		describe('number primitives', () => {
+			it('converts u8', () => {
+				const z = new u8(kusamaRegistry, 0);
+				expect(sanitizeNumbers(z)).toBe('0');
+
+				const m = new u8(kusamaRegistry, MAX_U8);
+				expect(sanitizeNumbers(m)).toBe(MAX_U8);
+			});
+
+			it('converts i8', () => {
+				const z = new I8(kusamaRegistry, 0);
+				expect(sanitizeNumbers(z)).toBe('0');
+
+				const min = new I8(kusamaRegistry, MIN_I8);
+				expect(sanitizeNumbers(min)).toBe(MIN_I8);
+
+				const max = new I8(kusamaRegistry, MAX_I8);
+				expect(sanitizeNumbers(max)).toBe(MAX_I8);
+			});
+
+			it('converts u16', () => {
+				const z = new U16(kusamaRegistry, 0);
+				expect(sanitizeNumbers(z)).toBe('0');
+
+				const max = new U16(kusamaRegistry, MAX_U16);
+				expect(sanitizeNumbers(max)).toBe(MAX_U16);
+			});
+
+			it('converts i16', () => {
+				const z = new I16(kusamaRegistry, 0);
+				expect(sanitizeNumbers(z)).toBe('0');
+
+				const min = new I16(kusamaRegistry, MIN_I16);
+				expect(sanitizeNumbers(min)).toBe(MIN_I16);
+
+				const max = new I16(kusamaRegistry, MAX_I16);
+				expect(sanitizeNumbers(max)).toBe(MAX_I16);
```

Comment:
It's weird that `new I16()` worked when your `MAX_I16` was actually `MAX_U16`. @jacogr, is this something the `I16` constructor should check for?

Reply:
It should yes. So there is def. an unsigned check missing (as I recall we did do have a total bits check, but this assumes only unsigned, so in this case it would pass since the number of bits are correct, even if the value overflows. Which is not quite helpful when your code starts misbehaving)